scripts/validate_docstrings.py

```diff
@@ -408,22 +409,53 @@ def validate_error_codes(errors):
     assert not nonexistent_errors, f"{nonexistent_errors} don't exist."
 
 
-def main(func_name, prefix, errors, output_format, ignore_deprecated, ignore_functions):
+def main(
+    func_name,
+    output_format,
+    prefix,
+    errors,
+    ignore_deprecated,
+    for_error_ignore_functions
+):
     """
     Main entry point. Call the validation for one or for all docstrings.
     """
+    if errors is None:
+        errors = []
+    if for_error_ignore_functions is None:
+        for_error_ignore_functions = {}
+    partial_validation_errors = for_error_ignore_functions.keys()
+    errors = [error for error in errors if error not in partial_validation_errors]
+
+    msg = []
+    if func_name is None:
+        if errors:
+            error_str = ", ".join(errors)
+            msg.append(f"Validate docstrings ({error_str})\n")
+        if partial_validation_errors:
+            error_str = ", ".join(partial_validation_errors)
+            msg.append(f"Partially validate docstrings ({error_str})\n")
+    else:
+        msg.append(f"Validate docstring in function {func_name}\n")
+    msg = "and\n".join(msg)
+    sys.stdout.write(msg)
+
```

Comment:
I think `for_errors_ignore_functions` (or just `ignore_errors`) should be a dictionary with the format:
```
{"pandas.Series.dt.as_unit": ["PR01", "PR02", "GL08"],
 "pandas.Series.dt.to_period": "PR02", "RT07"],
 ...
```

I don't think all this code would be needed then.

Reply:
True, made some parts of the code less complex and parameter variant also removes some edge cases the are hard to reason about.

Note that I did a major reformatting of the `check_code.sh` to match this parameter variant. 

I used the old parameter values (that are basically identical to those in main) and reformatted them in `__main__()` [here](https://github.com/pandas-dev/pandas/blob/5bf704d807281a36f5ae6ce16664485392e25163/scripts/validate_docstrings.py#L513-L523).